implementations/rust/ockam/ockam_api/src/nodes/service/transport.rs

```diff
@@ -1,185 +1,351 @@
         };
-        Ok(res)
+
+        let status = TransportStatus::new(ApiTransport {
+            tt: TransportType::Tcp,
+            tm: TransportMode::Listen,
+            socket_address: listener.socket_address(),
+            worker_address: "<none>".into(),
+            processor_address: listener.address().to_string(),
+            flow_control_id: listener.flow_control_id().clone(),
+        });
+
+        Response::ok(req.id()).body(status)
     }
 
-    pub(super) async fn add_transport<'a>(
+    pub(super) async fn create_tcp_connection<'a>(
         &self,
         req: &Request<'_>,
         dec: &mut Decoder<'_>,
-    ) -> Result<ResponseBuilder<TransportStatus<'a>>> {
-        let mut node_manager = self.node_manager.write().await;
-        let CreateTransport { tt, tm, addr, .. } = dec.decode()?;
-
-        use {super::TransportType::*, TransportMode::*};
+        ctx: &Context,
+    ) -> Result<ResponseBuilder<TransportStatus>> {
+        let node_manager = self.node_manager.read().await;
+        let CreateTcpConnection { addr, .. } = dec.decode()?;
 
-        info!(
-            "Handling request to create a new transport: {}, {}, {}",
-            tt, tm, addr
-        );
+        info!("Handling request to create a new TCP connection: {}", addr);
         let socket_addr = addr.to_string();
 
-        // TODO: Support FlowControls from ockam_command CLI
-        let flow_control_id = {
-            let s_addr: SocketAddr = socket_addr
-                .parse()
-                .map_err(|_| ApiError::generic("InvalidAddress tcp address"))?;
-
-            if s_addr.ip().is_loopback() {
-                // TODO: Enable FlowControl for loopback addresses as well
-                None
-            } else {
-                Some(node_manager.flow_controls.generate_id())
-            }
-        };
-        let res = match (tt, tm) {
-            (Tcp, Listen) => {
-                let options = if let Some(flow_control_id) = flow_control_id {
-                    TcpListenerOptions::as_spawner(&node_manager.flow_controls, &flow_control_id)
-                } else {
-                    TcpListenerOptions::insecure()
-                };
+        let options = TcpConnectionOptions::new();
 
-                node_manager
-                    .tcp_transport
-                    .listen(&addr, options)
-                    .await
-                    .map(|(socket, worker_address)| (socket.to_string(), worker_address))
-            }
-            (Tcp, Connect) => {
-                let options = if let Some(flow_control_id) = flow_control_id {
-                    TcpConnectionOptions::as_producer(&node_manager.flow_controls, &flow_control_id)
-                } else {
-                    TcpConnectionOptions::insecure()
-                };
+        // Add all Hop workers as consumers for Demo purposes
+        // Production nodes should not run any Hop workers
+        for hop in node_manager.registry.hop_services.keys() {
```

Comment:
Probably out of the scope of this PR, but maybe we should move all the `hop` service related code behind a feature flag to avoid leaking this funcionality into production.

Reply:
That's discussed